C++11 flag added and pystr_startswith implemented

andretadeu opened the pull request:
I added the PKG_CXXFLAGS=-std=c++11 in Makefile since it is not default in GCC 5.3. GCC 6 uses c++14 as default.

I implemented pystr_startswith using the same idea used in pystr_endswith.

andretadeu committed:
Add PKG_CXXFLAGS in Makefile

andretadeu committed:
Replace pystr_endswith implementation or one in C++.

andretadeu committed:
Update CONTRIBUTING.md

Ironholds commented:
If we're relying on C++11 for this set of patches, we should revert them (I
haven't had a chance to review the code and so didn't know). C++11 support
has _just_ showed up on Windows and isn't entirely present on the current
Ubuntu LTS; we're an age away from being in a space where this doesn't cut
out a heck of a lot of users.

On Mon, May 16, 2016 at 4:54 PM, André Tadeu de Carvalho <
notifications@github.com> wrote:

> I added the PKG_CXXFLAGS=-std=c++11 in Makefile since it is not default in
> GCC 5.3. GCC 6 uses c++14 as default.
> 
> ## I implemented pystr_startswith using the same idea used in pystr_endswith.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/nicolewhite/pystr/pull/9
> Commit Summary
> - Add PKG_CXXFLAGS in Makefile
> - Replace pystr_endswith implementation or one in C++.
> - Update CONTRIBUTING.md
> 
> File Changes
> - _M_ CONTRIBUTING.md
>   https://github.com/nicolewhite/pystr/pull/9/files#diff-0 (2)
> - _M_ Makefile
>   https://github.com/nicolewhite/pystr/pull/9/files#diff-1 (2)
> - _M_ R/RcppExports.R
>   https://github.com/nicolewhite/pystr/pull/9/files#diff-2 (4)
> - _M_ R/pystr_startswith.R
>   https://github.com/nicolewhite/pystr/pull/9/files#diff-3 (10)
> - _M_ src/RcppExports.cpp
>   https://github.com/nicolewhite/pystr/pull/9/files#diff-4 (14)
> - _A_ src/index_util.cpp
>   https://github.com/nicolewhite/pystr/pull/9/files#diff-5 (7)
> - _A_ src/index_util.h
>   https://github.com/nicolewhite/pystr/pull/9/files#diff-6 (8)
> - _M_ src/pystr_endswith.cpp
>   https://github.com/nicolewhite/pystr/pull/9/files#diff-7 (19)
> - _A_ src/pystr_startswith.cpp
>   https://github.com/nicolewhite/pystr/pull/9/files#diff-8 (54)
> - _A_ src/string_util.cpp
>   https://github.com/nicolewhite/pystr/pull/9/files#diff-9 (12)
> - _A_ src/string_util.h
>   https://github.com/nicolewhite/pystr/pull/9/files#diff-10 (9)
> - _M_ tests/testthat/test-pystr_startswith.R
>   https://github.com/nicolewhite/pystr/pull/9/files#diff-11 (19)
> 
> Patch Links:
> - https://github.com/nicolewhite/pystr/pull/9.patch
> - https://github.com/nicolewhite/pystr/pull/9.diff
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> https://github.com/nicolewhite/pystr/pull/9

andretadeu commented:
Are you able to compile using c++0x? This code also compiles using this version.

nicolewhite commented:
My bad, I didn't realize the implications. Is there another way to handle
Unicode in C++?

On Mon, May 16, 2016, 6:32 PM André Tadeu de Carvalho <
notifications@github.com> wrote:

> Are you able to compile using c++0x? This code also compiles using this
> version.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> https://github.com/nicolewhite/pystr/pull/9#issuecomment-219579041

andretadeu committed:
Set -std flag to c++0x.
Convert pystr_join implementation to C++.
Convert pystr_fill implementation to C++.

andretadeu committed:
pystr_find implemented in C++.

andretadeu committed:
Add some tests for unicode strings in pystr_upper and pystr_lower.
